bin/prepare.js

```diff
@@ -21,67 +29,111 @@ by creating separate databases for them
 and running their local bin/prepare.js scripts.
 `)
 
-async function main () {
-    let project, useHttps
+async function prepare () {
     program.parse()
-    const cliOpts = program.opts()
-
-    if (cliOpts.https) {
-        project = undefined
-        useHttps = 'app.localhost'
-    }
-
-    if (!project) project = 'local'
-
-    const secure = Boolean(useHttps)
-    const localhostDomain = (useHttps) ? useHttps : 'localhost'
-    let domains = `127.0.0.1 ${localhostDomain}`
+    const { https, filter } = program.opts()
 
+    // Step 1. Sanity checks
     await checkDockerComposePostgresIsRunning()
-    if (secure) await checkMkCertCommandAndLocalCerts(KEY_FILE, CERT_FILE, useHttps)
-
-    let apps = await getAllActualApps()
-    const appsIndexes = Object.assign({}, ...apps.map((appName, idx) => ({ [appName]: idx + 1 })))
-    if (cliOpts.filter) {
-        apps = apps.filter(app => cliOpts.filter.includes(app))
+    if (https) await checkMkCertCommandAndLocalCerts(KEY_FILE, CERT_FILE, DEFAULT_APP_HTTPS_SUBDOMAIN)
+
+    // Step 2. Get list of available apps and assign ports / indexes / dbNames and so on
+    const allApps = await getAllActualApps()
+    const appsWithData = allApps.map((app, idx) => {
+        const appOrder = idx + 1
+        if (app.type === 'KS') {
+            const sport = 8000 + appOrder
+            const port = 4000 + appOrder
+            return {
+                ...app,
+                pgName: `${DEFAULT_DB_NAME_PREFIX}-${app.name}`,
+                redisIndex: 5 + appOrder,
+                port,
+                sport,
+                serviceUrl: https
+                    ? `https://${app.name}.${DEFAULT_APP_HTTPS_SUBDOMAIN}:${sport}`
+                    : `http://localhost:${port}`,
+            }
+        } else if (app.type === 'Next') {
+            // NOTE: Next applications now do not support either specifying a port via an environment variable or working over https without a provider
+            // TODO: Add a dev script wrapper to NextJS apps to support local https runtime
+            return {
+                ...app,
+                serviceUrl: 'http://localhost:3000',
+            }
+        } else {
+            throw new Error('Unknown app type!')
+        }
+    })
+
+    // Step 3.1. Copy global .env.example values to .env with no override
+    await fillGlobalEnvWithDefaultValues()
+    // Step 3.2. Extract domains .env information, like CONDO_DOMAIN, MY_APP_DOMAIN and put it global monorepo env
+    for (const app of appsWithData) {
+        const domainEnvKey = `${app.name.toUpperCase().replaceAll('-', '_')}_DOMAIN`
+        await updateGlobalEnvFile(domainEnvKey, app.serviceUrl)
```

Comment:
They does, BUT....Some of names are not aligned with this. (For example `CC_DOMAIN` in stands instead of `CALLCENTER_DOMAIN` in prepare). We should fix it outside this PR